src/ansys/fluent/visualization/graphics/graphics_windows.py

```diff
@@ -66,76 +46,40 @@ def __init__(self, grid: tuple = (1, 1)):
         self._graphics_objs = []
         self.window_id = None
 
-    def add_graphics(
-        self,
-        object,
-        position: tuple = (0, 0),
-        opacity: float = 1,
-        title: str = "",
-    ) -> None:
-        """Add data to a plot.
-
-        Parameters
-        ----------
-        object: GraphicsDefn
-            Object to plot as a sub-plot.
-        position: tuple, optional
-            Position of the sub-plot.
-        opacity: float, optional
-            Transparency of the sub-plot.
-        title: str, optional
-            Title of the sub-plot (only for plots).
-        """
-        self._graphics_objs.append({**locals()})
-
-    def _all_plt_objs(self):
-        from ansys.fluent.core.post_objects.post_object_definitions import PlotDefn
-
-        for obj in self._graphics_objs:
-            if not isinstance(obj["object"].obj, PlotDefn):
-                return False
-        return True
-
-    def show(self) -> None:
+    def show(self, wind_id, renderer=None) -> None:
         """Render the objects in window and display the same."""
-        self.window_id = graphics_windows_manager._get_unique_window_id()
-        if self._all_plt_objs() and not pyviz.SINGLE_WINDOW:
-            p = PlotterWindow(grid=self._grid)
-            for obj in self._graphics_objs:
-                p.add_plots(obj["object"], position=obj["position"], title=obj["title"])
-            p.show(self.window_id)
-        else:
-            self.window_id = graphics_windows_manager.open_window(
-                window_id=self.window_id, grid=self._grid
+        self.window_id = wind_id
+        self.window_id = graphics_windows_manager.open_window(
+            window_id=self.window_id,
+            grid=self._grid,
+            renderer=renderer,
+        )
+        self.graphics_window = graphics_windows_manager._post_windows.get(
+            self.window_id
+        )
+        self._renderer = self.graphics_window.renderer
+        self.plotter = self.graphics_window.renderer.plotter
+        for i in range(len(self._graphics_objs)):
```

Comment:
`enumerate(...)` is a better option than `range(len(...))` in general. In some cases, `len` can be expensive, depending on the underlying iterable, while `enumerate` is more readable than a composite of two functions.

Reply:
Done, Thank you.